src/binaryManager.ts

```diff
@@ -93,33 +100,76 @@ async function getConfiguredMirrordBinary(): Promise<string | null> {
 }
 
 /**
- * Downloads mirrord binary (if needed) and returns its path
- */
-export async function getMirrordBinary(): Promise<string> {
+ * Toggles auto-update of mirrord binary.
+ * Criteria for auto-update:
+ * - Auto-update is enabled by default
+ * - if mirrord binary path is mentioned in workspace settings, then that is used
+ * - if a version is specified, that version is downloaded
+ * - if auto-update is enabled, then latest supported version is downloaded
+ * - if auto-update is disabled, any local mirrord binary is used
+ * 
+ * @param extensionActivate If true, then a global state is set so that any changes to the workspace settings pertaining
+ * to mirrord binary auto-update will prompt the user to reload the window.
+ * @returns Path to mirrord binary
+*/
+export async function getMirrordBinary(extensionActivate?: boolean): Promise<string> {
     const configured = await getConfiguredMirrordBinary();
+
     if (configured) {
+        vscode.window.showInformationMessage(`Using mirrord binary specified in settings: ${configured}`);
         return configured;
     }
+    const latestVersion = await getLatestSupportedVersion(10000);
+    const autoUpdateConfigured = vscode.workspace.getConfiguration().get("mirrord.autoUpdate");
+    const extensionPath = getExtensionMirrordPath().fsPath;
 
-    let foundLocal = await getLocalMirrordBinary();
-    // timeout is 1s if we have alternative or 10s if we don't
-    let timeout = foundLocal ? 1000 : 10000;
-    const latestVersion = await getLatestSupportedVersion(timeout);
-
-    // See if maybe we have it installed already, in correct version.
-    const localMirrord = await getLocalMirrordBinary(latestVersion);
-    if (localMirrord) {
-        const api = new MirrordAPI(localMirrord);
-        const installedVersion = await api.getBinaryVersion();
-        if (installedVersion === latestVersion) {
-            return localMirrord;
+    // if extension is activating, then set a global state to what was read in the workspace settings
+    if (extensionActivate) {
+        globalContext.workspaceState.update('autoUpdate', autoUpdateConfigured);
+    } else {
+        // extension is active, check if auto-update setting has changed
+        const autoUpdate = globalContext.workspaceState.get('autoUpdate');
+        if (autoUpdate !== autoUpdateConfigured) {
+            await new NotificationBuilder()
+                .withMessage("mirrord binary auto-update setting has changed, please reload the window for the change to take effect.")
+                .withGenericAction("Reload", async () => {
+                    await vscode.commands.executeCommand("workbench.action.reloadWindow");
+                })
+                .warning();
         }
+        return extensionPath;
     }
 
-    const extensionMirrordPath = getExtensionMirrordPath();
-    await downloadMirrordBinary(extensionMirrordPath, latestVersion);
+    if (typeof autoUpdateConfigured === 'string') {
```

Comment:
Document this please, i.e. possible values for autoUpdate

Reply:
do you want me to add docstring for it? its already documented in package.json as json schema

Reply:
It should be clear when reading the code without referring to package.json